Add ipadm_addr module

xen0l opened the pull request:
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New Module Pull Request

##### COMPONENT NAME
ipadm_addr

##### ANSIBLE VERSION
```
2.3
```

##### SUMMARY
Module for managing IP address configuration on illumos/Solaris systems.

xen0l committed:
Add ipadm_addr module

privateip commented on lib/ansible/modules/network/illumos/ipadm_addr.py:
```diff
@@ -0,0 +1,373 @@
+    module.exit_json(**result)
+
+
+from ansible.module_utils.basic import *
```
please use explicit imports

privateip commented on lib/ansible/modules/network/illumos/ipadm_addr.py:
```diff
@@ -0,0 +1,373 @@
+
+
+from ansible.module_utils.basic import *
+main()
```
please wrap in `if` block

privateip reviewed (changes requested):
* missing RETURN doc string
* missing METADATA
* please consider adding unit test cases

xen0l committed:
Add ANSIBLE_METADATA
Document return values
Make imports explicit
Use native YAML syntax in EXAMPLES

xen0l commented:
ready_for_review

xen0l commented:
@privateip I addressed you comments. As for unit tests, they will be shared among all modules to a certain degree (I am also speaking about modules, which are already included with Ansible). For that reason, I would like to add them in the next round if possible.

privateip commented on lib/ansible/modules/network/illumos/ipadm_addr.py:
```diff
@@ -0,0 +1,411 @@
+    - Create/delete static/dynamic IP addresses on network interfaces on Solaris/illumos systems.
+    - Up/down static/dynamic IP addresses on network interfaces on Solaris/illumos systems.
+    - Manage IPv6 link-local addresses on network interfaces on Solaris/illumos systems.
+version_added: "2.1"
```
VERSION_ADDED should be "2.3"

xen0l committed:
Fix version to 2.3

xen0l commented:
@privateip I bumped version to 2.3
